components/butterfly-test/src/lib.rs

```diff
@@ -122,12 +122,13 @@ impl DerefMut for SwimNet {
 
 impl SwimNet {
     pub fn new_with_suitability(suitabilities: Vec<u64>) -> SwimNet {
-        let count = suitabilities.len();
-        let mut members = Vec::with_capacity(count);
-        for x in 0..count {
-            members.push(start_server(&format!("{}", x), None, suitabilities[x]));
+        SwimNet {
+            members: suitabilities
+                .into_iter()
+                .enumerate()
+                .map(|(x, suitability)| start_server(&format!("{}", x), None, suitability))
+                .collect(),
```

Comment:
This is a bit of a refactor, but I'm pretty sure it's correct and more idiomatic

Reply:
Nice one!